### DISCUSSION TITLE
Callback functions in plugin, not theme?

### DISCUSSION THREAD

**Discussion by texorama** [TCattd/HTMX-API-WP#11] [General]
Hi, I want to use htmx in my plugin, and for that it's kind of awkward to put the call back functions "....htmx.php" in the themes folder. 
How can I ship those callbacks as part of my plugin? 
Thank you!

**Comment by EstebanForge** [comment]
Hi @texorama 

That can be already done. Check this out, please: https://github.com/TCattd/HTMX-API-WP/discussions/5
It's that what you need?

(I know docs are a pending task here. Any help it's appreciated)

Esteban

**Comment by texorama** [comment]
that's funny, I had JUST found this and fixed it that way:
```php
# for enabling htmx templates in our own codebase:
add_filter( 'hxwp/get_template_file/templates_path', array( $this, 'htmx_local_templates' ), 10, 1 );
function htmx_local_templates( $templates_path ): string {
		// if the template name ends with _sx.htmx.php, then we want to load the template from the htmx-templates
		// directory
		if ( str_ends_with($_SERVER['REQUEST_URI'], '_sx' ) ) {
			$templates_path = plugin_dir_path( __FILE__ ) . 'htmx-templates/';
		}

		return $templates_path;
}
```
(I didn't want to interfere with how the rest of the code might handle htmx, so I'm changing only if the request ends in _sx which is specific to my plugin.)

Thank you Esteban!

**Reply by EstebanForge** [reply]
Nice, @texorama 

I will put something about this filter on the main README on the repo. That's the best, until we get (or i get) time to create a proper documentation for the plugin.

Have a great day!
